.gitlab-ci.yml

```diff
@@ -0,0 +1,152 @@
+    - config_code=0
+    - python scripts/config-build.py -hc host-configs/LLNL/${HOST_CONFIG} -bt Release -bp build -DENABLE_DOXYGEN=OFF 2>&1 | tee ${SYSTEM}/${SYSTEM}_configure.log || config_code=$?
+    - if [[ $config_code -ne 0 ]]; then mv ${SYSTEM}/${SYSTEM}_configure.log ${SYSTEM}/${SYSTEM}_configure_FAILURE.log; else mv ${SYSTEM}/${SYSTEM}_configure.log ${SYSTEM}/${SYSTEM}_configure_SUCCESS.log; fi
+    - echo "~~~~~~~~~~ END - configure ~~~~~~~~~~~~~"
+
+    # BUILD
+    - echo "~~~~~~~~~~ START - build ~~~~~~~~~~~"
+    - build_code=0
+    - ${ALLOC_COMMAND} make VERBOSE=1 -C build -j ${NPROC} 2>&1 | tee ${SYSTEM}/${SYSTEM}_build.log || build_code=$?
+    - if [[ $build_code -ne 0 ]]; then mv ${SYSTEM}/${SYSTEM}_build.log ${SYSTEM}/${SYSTEM}_build_FAILURE.log; else mv ${SYSTEM}/${SYSTEM}_build.log ${SYSTEM}/${SYSTEM}_build_SUCCESS.log; fi
+    - echo "~~~~~~~~~~ END - build ~~~~~~~~~~~~~"
+
+    # UNIT TEST
+    - echo "~~~~~~~~~~ START - unit tests ~~~~~~~~~~~"
+    - unit_code=0
+    - cd build
+    - ${ALLOC_COMMAND} ctest --output-on-failure  2>&1 | tee ../${SYSTEM}/${SYSTEM}_unit_test.log || unit_code=$?
+    - cd ..
+    - if [[ $unit_code -ne 0 ]]; then mv ${SYSTEM}/${SYSTEM}_unit_test.log ${SYSTEM}/${SYSTEM}_unit_test_FAILURE.log; else mv ${SYSTEM}/${SYSTEM}_unit_test.log ${SYSTEM}/${SYSTEM}_unit_test_SUCCESS.log; fi
+    - echo "~~~~~~~~~~ END - unit tests ~~~~~~~~~~~~~"
+
+    # INTEGRATED TEST
+    - echo "~~~~~~~~~~ START - integrated tests ~~~~~~~~~~~"
+    - ./build/geosxats.sh -a veryclean
+    - integrated_code=0
+    - ${INTEGRATED_ALLOC_COMMAND} ./build/geosxats.sh --failIfTestsFail 2>&1 | tee ${SYSTEM}/${SYSTEM}_integrated_test.log || integrated_code=$?
+    - if [[ $integrated_code -ne 0 ]]; then mv ${SYSTEM}/${SYSTEM}_integrated_test.log ${SYSTEM}/${SYSTEM}_integrated_test_FAILURE.log; else mv ${SYSTEM}/${SYSTEM}_integrated_test.log ${SYSTEM}/${SYSTEM}_integrated_test_SUCCESS.log; fi
+    - mkdir ${SYSTEM}/integratedTestsDebugFiles
+    - cp -rL build/integratedTests/TestLogs ${SYSTEM}/
+    - grep -r -l --include=*.data "Error" build/integratedTests/ | xargs cp -t ${SYSTEM}/integratedTestsDebugFiles || true
+    - grep -r -l --include=*.err --exclude-dir="TestLogs*" "error" build/integratedTests/ | xargs cp -t ${SYSTEM}/integratedTestsDebugFiles || true
+    - echo "~~~~~~~~~~ END - integrated tests ~~~~~~~~~~~~~"
+
+    # CLEANUP + PRINT SUCCESS/FAILURES
+    - echo "~~~~~~~~~~ START - cleanup ~~~~~~~~~~~"
+    - ./build/geosxats.sh -a veryclean
+    - rm -rf build
+    - if [[ $config_code -ne 0 ]]; then echo "Configuration failed"; else echo "Configuration succeeded"; fi
+    - if [[ $build_code -ne 0 ]]; then echo "Build failed"; else echo "Build succeeded"; fi
+    - if [[ $unit_code -ne 0 ]]; then echo "Unit tests failed"; else echo "Unit tests succeeded"; fi
+    - if [[ $integrated_code -ne 0 ]]; then echo "Integrated tests failed"; else echo "Integrated tests succeeded"; fi
```

Comment:
Do you really wish to continue the process if some initial step failed?

Reply:
Yes, I think this is fine. 
The landing page of the nightlyTests (example for now: https://github.com/bmhan12/test) shows you which step failed first, so it should be clear where to go to debug.